Add index_configs field to log bucket configuration

mudim opened the pull request:
Ability to manage log bucket index configurations by adding `index_configs` field to logging_bucket_config resources.

Addresses feature request raised in [hashicorp/terraform-provider-google#13287](https://github.com/hashicorp/terraform-provider-google/issues/13287)

```release-note:enhancement
logging: added `index_configs` field to `logging_bucket_config` resource
```

```release-note:enhancement
logging: added `index_configs` field to `logging_project_bucket_config` resource
```

modular-magician commented:
Hello! I am a robot. It looks like you are a: Community Contributor ~Googler~ ~Core Contributor~. Tests will require approval to run.

@rileykarson, a repository maintainer, has been assigned to review your changes. If you have not received review feedback within 2 business days, please leave a comment on this PR asking them to take a look.

You can help make sure that review is quick by [doing a self-review](https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/) and by [running impacted tests locally](https://googlecloudplatform.github.io/magic-modules/get-started/run-provider-tests/).

mudim commented:
Hi @rileykarson, following the previous comment from MM bot, as it's been more than 2 days, wanted to follow up on review regarding this PR. Appreciate your time. Thanks

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_bucket_config.go:
```diff
@@ -88,6 +90,35 @@ See [Enabling CMEK for Logging Buckets](https://cloud.google.com/logging/docs/ro
+Note that some paths are automatically indexed, and other paths are not eligible for indexing. See [indexing documentation]( https://cloud.google.com/logging/docs/view/advanced-queries#indexed-fields) for details.
+For example: jsonPayload.request.status`,
+				},
+				"create_time": &schema.Schema{
```
We should leave this field off if not absolutely necessary `schema.TypeSet` does not play well with `Computed` fields, due to how diffs are calculated for sets, and Terraform tries to reuse old computed values.

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_bucket_config.go:
```diff
@@ -405,3 +458,40 @@ func flattenCmekSettings(cmekSettings interface{}) []map[string]interface{} {
+	transformedIndexConfigs := make([]map[string]interface{}, 0, len(originalIndexConfigs))
+	for _, entry := range originalIndexConfigs {
+		original := entry.(map[string]interface{})
+		var validIndexType = false
```
Can we just defer to the API error here? We're already in apply, let's not add extra complexity clientside.

If this was a ValidateFunc- which works at plan time- it may be worth keeping around.

rileykarson commented on mmv1/third_party/terraform/website/docs/r/logging_folder_bucket_config.html.markdown:
````diff
@@ -28,6 +28,22 @@ resource "google_logging_folder_bucket_config" "basic" {
 
+Create logging bucket with index configs
+
+```hcl
````
Nit: spacing on these blocks. They should be 2-spaced spaces and are currently mixed tabs+spaces.

(All tabs would also work for consistency within the file, but the other entry in this file is also wrong)

rileykarson commented on mmv1/third_party/terraform/website/docs/r/logging_folder_bucket_config.html.markdown:
```diff
@@ -42,6 +58,19 @@ The following arguments are supported:
+* `field_path` - The LogEntry field path to index.
+  Note that some paths are automatically indexed, and other paths are not eligible for indexing. See [indexing documentation]( https://cloud.google.com/logging/docs/view/advanced-queries#indexed-fields) for details.
+
+* `type` - The type of data in this index.
```
We may want to list enum values here!

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Breaking Change(s) Detected
The following breaking change(s) were detected within your pull request.

* Field `index_configs.field_path` changed from optional to required on `google_logging_billing_account_bucket_config` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes#field-optional-to-required)
* Field `index_configs.field_path` changed from optional to required on `google_logging_folder_bucket_config` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes#field-optional-to-required)
* Field `index_configs.field_path` changed from optional to required on `google_logging_organization_bucket_config` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes#field-optional-to-required)
* Field `index_configs.field_path` changed from optional to required on `google_logging_project_bucket_config` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes#field-optional-to-required)
* Field `index_configs.type` changed from optional to required on `google_logging_billing_account_bucket_config` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes#field-optional-to-required)
* Field `index_configs.type` changed from optional to required on `google_logging_folder_bucket_config` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes#field-optional-to-required)
* Field `index_configs.type` changed from optional to required on `google_logging_organization_bucket_config` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes#field-optional-to-required)
* Field `index_configs.type` changed from optional to required on `google_logging_project_bucket_config` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes#field-optional-to-required)


If you believe this detection to be incorrect please raise the concern with your reviewer. If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window. An `override-breaking-change` label can be added to allow merging.

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9109-old..auto-pr-9109) ( 6 files changed, 294 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9109-old..auto-pr-9109) ( 6 files changed, 294 insertions(+))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_logging_billing_account_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_billing_account_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```
Resource: `google_logging_folder_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_folder_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```
Resource: `google_logging_organization_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_organization_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```

rileykarson commented:
Robot seems wrong here, ignore the breaking change detector / missing test messages from it.

modular-magician commented:
#### Tests analytics 
Total tests: `3111` 
Passed tests `2778` 
Skipped tests: `314` 
Affected tests: `19` 

#### Action taken 
<details> <summary>Found 19 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccContainerNodePool_withUpgradeSettings|TestAccContainerNodePool_withKubeletConfig|TestAccContainerCluster_withAddons|TestAccLoggingMetric_loggingMetricLoggingBucketExample|TestAccLoggingLogView_loggingLogViewLongNameExample|TestAccLoggingLogView_loggingLogViewBasicExampleUpdate|TestAccLoggingBucketConfigOrganization_basic|TestAccLoggingBucketConfigBillingAccount_basic|TestAccLoggingBucketConfigFolder_basic|TestAccLoggingBucketConfigProject_locked|TestAccLoggingBucketConfigProject_analyticsEnabled|TestAccLoggingBucketConfigProject_basic|TestAccLoggingBucketConfigProject_cmekSettings|TestAccLoggingBucketConfigProject_indexConfigs|TestAccLoggingMetric_loggingBucket|TestAccLoggingLinkedDataset_loggingLinkedDatasetBasicExample|TestAccLoggingLinkedDataset_loggingLinkedDatasetAllParamsExample|TestAccLoggingLogView_loggingLogViewBasicExample|TestAccLoggingBucketConfig_CreateBuckets_withCustomId </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccLoggingMetric_loggingMetricLoggingBucketExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingMetric_loggingMetricLoggingBucketExample.log)]
`TestAccLoggingLogView_loggingLogViewLongNameExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingLogView_loggingLogViewLongNameExample.log)]
`TestAccLoggingLogView_loggingLogViewBasicExampleUpdate`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingLogView_loggingLogViewBasicExampleUpdate.log)]
`TestAccLoggingBucketConfigOrganization_basic`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingBucketConfigOrganization_basic.log)]
`TestAccLoggingBucketConfigFolder_basic`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingBucketConfigFolder_basic.log)]
`TestAccLoggingBucketConfigProject_analyticsEnabled`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingBucketConfigProject_analyticsEnabled.log)]
`TestAccLoggingBucketConfigProject_basic`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingBucketConfigProject_basic.log)]
`TestAccLoggingBucketConfigProject_cmekSettings`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingBucketConfigProject_cmekSettings.log)]
`TestAccLoggingBucketConfigProject_indexConfigs`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingBucketConfigProject_indexConfigs.log)]
`TestAccLoggingMetric_loggingBucket`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingMetric_loggingBucket.log)]
`TestAccLoggingLinkedDataset_loggingLinkedDatasetBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingLinkedDataset_loggingLinkedDatasetBasicExample.log)]
`TestAccLoggingLinkedDataset_loggingLinkedDatasetAllParamsExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingLinkedDataset_loggingLinkedDatasetAllParamsExample.log)]
`TestAccLoggingLogView_loggingLogViewBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingLogView_loggingLogViewBasicExample.log)]
`TestAccLoggingBucketConfig_CreateBuckets_withCustomId`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingBucketConfig_CreateBuckets_withCustomId.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withUpgradeSettings`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/build-log/recording_build/TestAccContainerNodePool_withUpgradeSettings_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccContainerNodePool_withUpgradeSettings.log)]
`TestAccContainerNodePool_withKubeletConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/build-log/recording_build/TestAccContainerNodePool_withKubeletConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccContainerNodePool_withKubeletConfig.log)]
`TestAccContainerCluster_withAddons`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/build-log/recording_build/TestAccContainerCluster_withAddons_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccContainerCluster_withAddons.log)]
`TestAccLoggingBucketConfigBillingAccount_basic`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/build-log/recording_build/TestAccLoggingBucketConfigBillingAccount_basic_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingBucketConfigBillingAccount_basic.log)]
`TestAccLoggingBucketConfigProject_locked`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/build-log/recording_build/TestAccLoggingBucketConfigProject_locked_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording/TestAccLoggingBucketConfigProject_locked.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/7b74ce6f-36c4-4001-b8c0-3c209dc73efb/recording) for each test

mudim commented:
Hi @rileykarson, thanks for taking time and review on it. Have addressed the comments raised above. 

One thing around the tests reported by MM as failures, ignoring the unrelated ones to this change, have looked at the 2 logging tests reported as failed. However, they pass locally both for ga and beta versions. Any feedback on that part would be helpful. 

Thanks

rileykarson commented:
> TestAccLoggingBucketConfigBillingAccount_basic

This was a missing permission. It was in our manual bootstrapping guide but not applied to the billing account of our presubmits (which previously replayed logs from an old target).

> TestAccLoggingBucketConfigProject_locked

Failing at HEAD, filed https://github.com/hashicorp/terraform-provider-google/issues/16099

rileykarson commented on mmv1/third_party/terraform/website/docs/r/logging_organization_bucket_config.html.markdown:
````diff
@@ -27,6 +27,22 @@ resource "google_logging_organization_bucket_config" "basic" {
 }
 ```
 
+Create logging bucket with index configs
````
nit: Let's just roll the new field into the entry above.

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_bucket_config_test.go:
```diff
@@ -523,3 +523,69 @@ func getLoggingBucketConfigs(context map[string]interface{}) map[string]string {
 
 }
+
+func TestAccLoggingBucketConfigProject_indexConfigs(t *testing.T) {
```
Can you add a test on one of the other targets? It's not a big deal if you can't run it locally, since I'll be able to in our CI.

rileykarson reviewed (commented):
Sorry, forgot to click "Submit review" yesterday!

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 300 insertions(+), 11 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 300 insertions(+), 11 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_logging_billing_account_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_billing_account_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```
Resource: `google_logging_folder_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_folder_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```
Resource: `google_logging_organization_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_organization_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3120` 
Passed tests `2799` 
Skipped tests: `315` 
Affected tests: `6` 

#### Action taken 
<details> <summary>Found 6 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccContainerNodePool_withKubeletConfig|TestAccContainerNodePool_withUpgradeSettings|TestAccContainerCluster_withAddons|TestAccLoggingBucketConfigBillingAccount_basic|TestAccLoggingBucketConfigProject_locked|TestAccWorkstationsWorkstationConfig_update </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccLoggingBucketConfigBillingAccount_basic`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/recording/TestAccLoggingBucketConfigBillingAccount_basic.log)]
`TestAccWorkstationsWorkstationConfig_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/recording/TestAccWorkstationsWorkstationConfig_update.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withKubeletConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/build-log/recording_build/TestAccContainerNodePool_withKubeletConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/recording/TestAccContainerNodePool_withKubeletConfig.log)]
`TestAccContainerNodePool_withUpgradeSettings`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/build-log/recording_build/TestAccContainerNodePool_withUpgradeSettings_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/recording/TestAccContainerNodePool_withUpgradeSettings.log)]
`TestAccContainerCluster_withAddons`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/build-log/recording_build/TestAccContainerCluster_withAddons_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/recording/TestAccContainerCluster_withAddons.log)]
`TestAccLoggingBucketConfigProject_locked`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/build-log/recording_build/TestAccLoggingBucketConfigProject_locked_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/recording/TestAccLoggingBucketConfigProject_locked.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/ff8a1df9-4c31-4571-9ab0-d6f4b00caef6/recording) for each test

mudim commented:
Thanks @rileykarson. Have pushed up changes to 
- Add a new test at org level
- Remove usage of `createTime` at resource read step
- Update markdown to collapse indexConfig into existing example

mudim commented:
Hi @rileykarson, just wanted to follow up on previous changes addressing comments, when u get a chance pls. Thanks

rileykarson commented:
Thanks! Running tests, sorry for the delay on my end.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 341 insertions(+), 15 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 341 insertions(+), 15 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_logging_billing_account_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_billing_account_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```
Resource: `google_logging_folder_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_folder_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3148` 
Passed tests `2827` 
Skipped tests: `316` 
Affected tests: `5` 

#### Action taken 
<details> <summary>Found 5 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccAlloydbBackup_alloydbBackupFullExample|TestAccAlloydbCluster_secondaryClusterMandatoryFields|TestAccLoggingBucketConfigOrganization_indexConfigs|TestAccLoggingBucketConfigProject_locked|TestAccLoggingProjectSink_updatePreservesCustomWriter </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccLoggingBucketConfigOrganization_indexConfigs`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/recording/TestAccLoggingBucketConfigOrganization_indexConfigs.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccAlloydbBackup_alloydbBackupFullExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/build-log/recording_build/TestAccAlloydbBackup_alloydbBackupFullExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/recording/TestAccAlloydbBackup_alloydbBackupFullExample.log)]
`TestAccAlloydbCluster_secondaryClusterMandatoryFields`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/build-log/recording_build/TestAccAlloydbCluster_secondaryClusterMandatoryFields_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/recording/TestAccAlloydbCluster_secondaryClusterMandatoryFields.log)]
`TestAccLoggingBucketConfigProject_locked`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/build-log/recording_build/TestAccLoggingBucketConfigProject_locked_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/recording/TestAccLoggingBucketConfigProject_locked.log)]
`TestAccLoggingProjectSink_updatePreservesCustomWriter`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/build-log/recording_build/TestAccLoggingProjectSink_updatePreservesCustomWriter_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/recording/TestAccLoggingProjectSink_updatePreservesCustomWriter.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/be9dadd2-712d-43f1-86fa-4b35292423b5/recording) for each test

mudim commented:
> Thanks! Running tests, sorry for the delay on my end.

No problem, thanks again for taking time with this 👍

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_bucket_config.go:
```diff
@@ -287,6 +314,13 @@ func resourceLoggingBucketConfigRead(d *schema.ResourceData, meta interface{}) e
 
+	indexConfigs, ok := res["indexConfigs"]
+	if ok {
+		if err := d.Set("index_configs", flattenIndexConfigs(indexConfigs.([]interface{}))); err != nil {
```
Ah- we want to set this all the time, or else we won't detect if the setting has been removed out of band. Terraform preserves the old value in state if you do nothing in Read.

You can safely remove the `ok`, the flattener will handle a `nil` appropriately.

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_project_bucket_config.go:
```diff
@@ -106,6 +106,35 @@ See [Enabling CMEK for Logging Buckets](https://cloud.google.com/logging/docs/ro
 			},
 		},
 	},
+	"index_configs": &schema.Schema{
```
This file needs `gofmt` run on it! `gofmt -w -s mmv1/third_party/terraform/services/logging/resource_logging_project_bucket_config.go` from the repo root should do it.

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_project_bucket_config.go:
```diff
@@ -106,6 +106,35 @@ See [Enabling CMEK for Logging Buckets](https://cloud.google.com/logging/docs/ro
+Note that some paths are automatically indexed, and other paths are not eligible for indexing. See [indexing documentation]( https://cloud.google.com/logging/docs/view/advanced-queries#indexed-fields) for details.
+For example: jsonPayload.request.status`,
+				},
+				"create_time": &schema.Schema{
```
Ah- same as https://github.com/GoogleCloudPlatform/magic-modules/pull/9109#discussion_r1343168254, `Computed` values and `TypeSet` don't play nicely. We should remove this as well (and any associated expander/flattener code, but I think those are already updated since they were shared)

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_bucket_config_test.go:
```diff
@@ -523,3 +523,129 @@ func getLoggingBucketConfigs(context map[string]interface{}) map[string]string {
+	description = "retention test 30 days"
+	bucket_id = "_Default"
+
+    index_configs {
```
```suggestion
	index_configs {
```

(it's probably easier to just make this change in your editor)

mudim replied:
Odd that it's showing correct format locally, will check again 👍

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_bucket_config_test.go:
```diff
@@ -523,3 +523,129 @@ func getLoggingBucketConfigs(context map[string]interface{}) map[string]string {
+				ImportStateVerifyIgnore: []string{"organization"},
+			},
+			{
+				Config: testAccLoggingBucketConfigOrganization_indexConfigs(context, "INDEX_TYPE_STRING", "INDEX_TYPE_STRING"),
```
Can we change one of the values here, similar to TestAccLoggingBucketConfigProject_indexConfigs below?

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_project_bucket_config.go:
```diff
@@ -282,6 +316,13 @@ func resourceLoggingProjectBucketConfigRead(d *schema.ResourceData, meta interfa
 	}
 
+	indexConfigs, ok := res["indexConfigs"]
+	if ok {
```
Same as above- we need to unconditionally write this so that we detect the removal of the value out-of-band.

rileykarson reviewed (commented):
Acceptance test failures are all unrelated. The build-and-test failures are just a formatting issue that `gofmt` will fix. I noticed I missed something in Read and that `create_time` was still around in project while scanning through.

mudim commented:
Thanks @rileykarson . Have pushed up changes addressing recent comments 👍

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_bucket_config.go:
```diff
@@ -287,6 +314,10 @@ func resourceLoggingBucketConfigRead(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("Error setting cmek_settings: %s", err)
 	}
 
+	if err := d.Set("index_configs", flattenIndexConfigs(indexConfigs.([]interface{}))); err != nil {
```
```suggestion
	if err := d.Set("index_configs", flattenIndexConfigs(res["indexConfigs"].([]interface{}))); err != nil {
```

rileykarson replied:
(I haven't run these at all- you may want to make the changes locally and ensure it builds / a test passes to potentially save a round trip!)

mudim replied:
Yeah, have been for all builds. Should've paid close attn to latest one as it was the cache being on that tripped this over. Should've looked close to see the test results were cached for above commit

rileykarson replied:
Oh, you're all good! I've just had a handful of cases like this turn into multiple day-long round trips of minor syntax changes enough times that I try to warn that I am suggesting a literal fix to a single syntax problem and not guaranteeing the code actually compiles or anything.

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_project_bucket_config.go:
```diff
@@ -282,6 +309,10 @@ func resourceLoggingProjectBucketConfigRead(d *schema.ResourceData, meta interfa
 		return fmt.Errorf("Error setting cmek_settings: %s", err)
 	}
 
+	if err := d.Set("index_configs", flattenIndexConfigs(indexConfigs.([]interface{}))); err != nil {
```
```suggestion
	if err := d.Set("index_configs", flattenIndexConfigs(res["indexConfigs"].([]interface{}))); err != nil {
```

mudim replied:
My bad, was rushing to get this change in. Resolved now

mudim commented:
`make test` at provider level is now successful with no errors reported.
ATs running fine with below output

command: 
`make testacc TEST=./google/services/logging TESTARGS='-run=TestAccLoggingBucketConfigProject_indexConfigs'`

result:
```TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google/services/logging -v -run=TestAccLoggingBucketConfigProject_indexConfigs -timeout 240m -ldflags="-X=http://github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN  TestAccLoggingBucketConfigProject_indexConfigs
=== PAUSE TestAccLoggingBucketConfigProject_indexConfigs
=== CONT TestAccLoggingBucketConfigProject_indexConfigs
--- PASS: TestAccLoggingBucketConfigProject_indexConfigs (43.29s)
PASS
ok 	http://github.com/hashicorp/terraform-provider-google/google/services/logging	47.747s

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 328 insertions(+), 15 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 328 insertions(+), 15 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_logging_billing_account_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_billing_account_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```
Resource: `google_logging_folder_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_folder_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```

modular-magician commented:
$\textcolor{red}{\textsf{The provider crashed while running the VCR tests in REPLAYING mode}}$ 
$\textcolor{red}{\textsf{Please fix it to complete your PR}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/48f40ae5-2eba-4128-be5e-487ccaa881a3/build-log/replaying_test.log)

rileykarson commented:
That doesn't seem right... let me try again.

/gcbrun

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 328 insertions(+), 15 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 328 insertions(+), 15 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_logging_billing_account_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_billing_account_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```
Resource: `google_logging_folder_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_folder_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```

modular-magician commented:
$\textcolor{red}{\textsf{The provider crashed while running the VCR tests in REPLAYING mode}}$ 
$\textcolor{red}{\textsf{Please fix it to complete your PR}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/bb245409-3ad9-4f5d-b109-475dcb45d5e7/build-log/replaying_test.log)

mahesh-hmrc commented:
Hi @rileykarson, hope all well. Just wanted to follow up on above and if there is anything I can/need to do on my side. Thanks

mudim commented:
Hi @rileykarson, hope all well. Just wanted to follow up on above and if there is anything I can/need to do on my side. Thanks

rileykarson commented:
I'm gonna try this one more time before investigating deeper, sorry for the delay!

/gcbrun

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 328 insertions(+), 15 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 328 insertions(+), 15 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_logging_billing_account_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_billing_account_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```
Resource: `google_logging_folder_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_folder_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```

modular-magician commented:
$\textcolor{red}{\textsf{The provider crashed while running the VCR tests in REPLAYING mode}}$ 
$\textcolor{red}{\textsf{Please fix it to complete your PR}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/4f8ecd62-9892-4fd3-b2f1-7f4dcd6ea485/build-log/replaying_test.log)

trodge commented on mmv1/third_party/terraform/services/logging/resource_logging_bucket_config.go:
```diff
@@ -287,6 +314,10 @@ func resourceLoggingBucketConfigRead(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("Error setting cmek_settings: %s", err)
 	}
 
+	if err := d.Set("index_configs", flattenIndexConfigs(res["indexConfigs"].([]interface{}))); err != nil {
```
I suggest breaking this line up and first checking that `res["indexConfigs"]` is, in fact, a slice of `any` (or `interface{}`).

rileykarson replied:
https://github.com/GoogleCloudPlatform/magic-modules/pull/9109#discussion_r1367251751 instead- the stanza we use in generated code like subnetwork is resilient to as many edge cases as possible.

rileykarson commented on mmv1/third_party/terraform/services/logging/resource_logging_bucket_config.go:
```diff
@@ -287,6 +314,10 @@ func resourceLoggingBucketConfigRead(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("Error setting cmek_settings: %s", err)
 	}
 
+	if err := d.Set("index_configs", flattenIndexConfigs(res["indexConfigs"].([]interface{}))); err != nil {
```
Ah, we're dying on this field when the value is `nil`. We'll want to change this to just `d.Set("index_configs", flattenIndexConfigs(res["indexConfigs"]));` and handle the value inside the flatten function. https://github.com/hashicorp/terraform-provider-google-beta/blob/35d78ff10768b60b27d020d08fde466d43826470/google-beta/services/compute/resource_compute_subnetwork.go#L1124-L1142 is a good reference.

mudim replied:
I see. Thanks for taking time with it, will get that updated now 👍

mudim commented:
Hi @rileykarson, have pushed up change to handle null index_configs during crud. Thanks

mudim commented:
Hi @rileykarson, hope all well. Just wanted to follow up on above and when you have few mins pls. Thanks

rileykarson commented:
Running generate+test! Apologies, had some other work superseding most of my time for reviews.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 324 insertions(+), 15 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9109-old..auto-pr-9109) ( 7 files changed, 324 insertions(+), 15 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_logging_billing_account_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_billing_account_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```
Resource: `google_logging_folder_bucket_config` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_logging_folder_bucket_config" "primary" {
  index_configs {
    field_path = # value needed
    type       = # value needed
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3186` 
Passed tests `2857` 
Skipped tests: `325` 
Affected tests: `4` 

#### Action taken 
<details> <summary>Found 4 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDataprocClusterIamPolicy|TestAccLoggingProjectSink_updatePreservesCustomWriter|TestAccLoggingBucketConfigProject_locked|TestAccLoggingBucketConfigOrganization_indexConfigs </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDataprocClusterIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/21a0cd38-8c20-4239-9173-a4a023baeb7c/recording/TestAccDataprocClusterIamPolicy.log)]
`TestAccLoggingBucketConfigOrganization_indexConfigs`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/21a0cd38-8c20-4239-9173-a4a023baeb7c/recording/TestAccLoggingBucketConfigOrganization_indexConfigs.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccLoggingProjectSink_updatePreservesCustomWriter`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/21a0cd38-8c20-4239-9173-a4a023baeb7c/build-log/recording_build/TestAccLoggingProjectSink_updatePreservesCustomWriter_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/21a0cd38-8c20-4239-9173-a4a023baeb7c/recording/TestAccLoggingProjectSink_updatePreservesCustomWriter.log)]
`TestAccLoggingBucketConfigProject_locked`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/21a0cd38-8c20-4239-9173-a4a023baeb7c/build-log/recording_build/TestAccLoggingBucketConfigProject_locked_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/21a0cd38-8c20-4239-9173-a4a023baeb7c/recording/TestAccLoggingBucketConfigProject_locked.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/21a0cd38-8c20-4239-9173-a4a023baeb7c/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9109/artifacts/21a0cd38-8c20-4239-9173-a4a023baeb7c/recording) for each test

mudim commented:
Hi @rileykarson, just wanted to follow up the last set of VCR test results. From what I remember one of them was linked to an [issue](https://github.com/hashicorp/terraform-provider-google/issues/16099) you've raised already. Are we blocked on that for the approval on this one? Thanks

trodge commented:
> Hi @rileykarson, just wanted to follow up the last set of VCR test results. From what I remember one of them was linked to an [issue](https://github.com/hashicorp/terraform-provider-google/issues/16099) you've raised already. Are we blocked on that for the approval on this one? Thanks

It shouldn't be blocking. The failing test is unrelated to this change, which is tested by `TestAccLoggingBucketConfigOrganization_indexConfigs`.

ScottSuarez commented:
This change appears to have broken tests in the GA and BETA projects. The API is returning default fields where none were set by the user. @rileykarson shall we revert?

```
=== CONT  TestAccLoggingBucketConfigOrganization_basic
    vcr_utils.go:152: Step 1/4 error: After applying this test step and performing a `terraform refresh`, the plan was not empty.
        stdout
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        Terraform will perform the following actions:
          # google_logging_organization_bucket_config.basic will be updated in-place
          ~ resource "google_logging_organization_bucket_config" "basic" {
                id              = "organizations/*******/locations/global/buckets/_Default"
                name            = "organizations/*******/locations/global/buckets/_Default"
                # (6 unchanged attributes hidden)
              - index_configs {
                  - field_path = "jsonPayload.request.url" -> null
                  - type       = "INDEX_TYPE_STRING" -> null
                }
              - index_configs {
                  - field_path = "jsonPayload.response.status" -> null
                  - type       = "INDEX_TYPE_INTEGER" -> null
                }
            }
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccLoggingBucketConfigOrganization_basic (13.55s)
FAIL

```

Our tests are failing in these locations 
https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_Google_NightlyTests_GOOGLE_PACKAGE_LOGGING/54819?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildTestsSection=true&expandBuildDeploymentsSection=false

ScottSuarez commented:
Revert here https://github.com/GoogleCloudPlatform/magic-modules/pull/9448

ScottSuarez commented:
Not reverting after talking with the team!
